readthedocs/doc_builder/environments.py

```diff
@@ -515,44 +498,14 @@ def handle_exception(self, exc_type, exc_value, _):
         """
         Exception handling for __enter__ and __exit__.
 
-        This reports on the exception we're handling and special cases
-        subclasses of BuildEnvironmentException. For
-        :py:class:`BuildEnvironmentWarning`, exit this context gracefully, but
-        don't mark the build as a failure. For all other exception classes,
+        If the exception's type is :py:class:`BuildEnvironmentWarning`,
+        exit this context gracefully. For all other exception classes,
         including :py:class:`BuildEnvironmentError`, the build will be marked as
         a failure and the context will be gracefully exited.
-
-        If the exception's type is :py:class:`BuildEnvironmentWarning` or it's
-        an exception marked as ``WARNING_EXCEPTIONS`` we log the problem as a
-        WARNING, otherwise we log it as an ERROR.
         """
         if exc_type is not None:
-            log_level_function = None
-            if issubclass(exc_type, BuildEnvironmentWarning):
-                log_level_function = log.warning
-            elif exc_type in self.WARNING_EXCEPTIONS:
-                log_level_function = log.warning
-                self.failure = exc_value
-            else:
-                log_level_function = log.error
+            if not issubclass(exc_type, BuildEnvironmentWarning):
                 self.failure = exc_value
```

Comment:
I think we do want to log the `BuildEnvironmentWarning` ones as a `log.warning`, but it seems that you remove all the `log.` calls.

Reply:
Yes.
I thought the `task.throws` is already logging them and then we again log them here, so it is kind of redundant.
However, I think that this PR is no longer required after #5118 .
Would like to know your views on this.